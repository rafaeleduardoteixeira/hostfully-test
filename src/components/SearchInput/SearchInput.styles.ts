import styled from 'styled-components';
import { colors, media } from '../../themes';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  flex: 1;
`;

export const SearchInputField = styled.input`
  height: 50px;
  border: none;
  outline: none;
  padding: 0 20px;
  font-size: 1rem;
  color: hsl(200, 5%, 40%);
  background-color: hsl(200, 5%, 95%);
  border-radius: 10px;

  &::placeholder {
    color: hsl(200, 5%, 80%);
  }

  &:focus {
    outline: ${colors.grayLight} auto 1px;
  }

  @media (${media.mobile}) {
    font-size: 0.9rem;
    height: 40px;
  }
`;
