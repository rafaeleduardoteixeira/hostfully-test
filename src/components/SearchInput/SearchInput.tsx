import { Container, SearchInputField } from './SearchInput.styles';

interface SearchInputProps {
  value?: string;
  onChange: (value: string) => void;
}

export const SearchInput = ({ value, onChange }: SearchInputProps) => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChange(event.target.value);
  };

  return (
    <Container>
      <SearchInputField value={value} onChange={handleChange} placeholder="Enter a property name or city" />
    </Container>
  );
};
