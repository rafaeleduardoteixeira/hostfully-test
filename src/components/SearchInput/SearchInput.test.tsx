import React, { act } from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { SearchInput } from './SearchInput';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('SearchInput', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<SearchInput onChange={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render SearchInput', () => {
    render(<SearchInput onChange={jest.fn()} />);
    const SearchInputElement = screen.getByTestId('SearchInput');
    expect(SearchInputElement).toBeInTheDocument();
  });

  test('Should call onChange when input value changes', () => {
    const onChange = jest.fn();
    render(<SearchInput onChange={onChange} />);
    const InputSearchElement = screen.getByPlaceholderText('Enter a property name or city');
    act(() => {
      fireEvent.change(InputSearchElement, { target: { value: 'test' } });
    });
    expect(onChange).toBeCalled();
  });
});
