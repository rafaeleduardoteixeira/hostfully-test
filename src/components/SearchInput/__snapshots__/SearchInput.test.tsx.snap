// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu hhgZVd"
    data-testid="SearchInput"
  >
    <input
      class="sc-gJhJTp hFeTUB"
      placeholder="Enter a property name or city"
    />
  </div>
</DocumentFragment>
`;
